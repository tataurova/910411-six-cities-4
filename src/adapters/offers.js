export const getOffer = (data) => {
  return {
    id: data.id,
    title: data.title,
    city: data.city.name,
    cityCoordinates: [data.city.location.latitude, data.city.location.longitude],
    cityZoom: data.city.location.zoom,
    coordinates: [data.location.latitude, data.location.longitude],
    type: data.type,
    price: data.price,
    rating: data.rating,
    premium: data.is_premium,
    favorite: data.is_favorite,
    photo: data.preview_image,
    bedrooms: data.bedrooms,
    adults: data.max_adults,
    additional: data.goods,
    details: data.description,
    owner: {
      photo: data.host.avatar_url,
      name: data.host.name,
      isSuper: data.host.is_pro,
    },
    photos: data.images,
    reviews: [ // todo reviews from server
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  };
};
