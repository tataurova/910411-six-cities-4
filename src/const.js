export const PLACES_COUNT = 312;

export const MapSettings = {
  city: [52.38333, 4.9],
  icon: {
    iconUrl: `img/pin.svg`,
    iconSize: [30, 30],
  },
  zoom: 12,
};
