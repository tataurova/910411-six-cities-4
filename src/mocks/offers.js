export default [
  {
    id: 1,
    title: `Canal View Prinsengracht`,
    city: `Cologne`,
    cityCoordinates: [50.938361, 6.959974],
    cityZoom: 13,
    coordinates: [50.960361, 6.967974],
    type: `Apartment`,
    price: 120,
    rating: 4,
    premium: true,
    photo: `img/apartment-01.jpg`,
    bedrooms: 3,
    adults: 6,
    additional: [`Wifi`, `Heating`, `Kitchen`],
    details: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
     The building is green and from 18th century. An independent House, strategically located between Rembrand
     Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    owner: {
      photo: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      isSuper: true,
    },
    photos: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/room.jpg`
    ],
    reviews: [
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  }, {
    id: 2,
    title: `Wood and stone place`,
    city: `Cologne`,
    cityCoordinates: [50.938361, 6.959974],
    cityZoom: 13,
    coordinates: [50.913361, 6.9509739999999995],
    type: `Private room`,
    price: 80,
    rating: 4,
    premium: false,
    photo: `img/room.jpg`,
    bedrooms: 2,
    adults: 4,
    additional: [`Wifi`, `Kitchen`, `Cable TV`, `Washing machine`],
    details: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
     The building is green and from 18th century. An independent House, strategically located between Rembrand
     Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    owner: {
      photo: `img/avatar-max.jpg`,
      name: `Max`,
      isSuper: false,
    },
    photos: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/room.jpg`
    ],
    reviews: [
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  }
];

/* eslint-disable */
export const serverOffers = [
  {
    id: 1,
    title: `Canal View Prinsengracht`,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    location: {
      latitude: 50.960361,
      longitude: 6.967974,
    },
    type: `Apartment`,
    price: 120,
    rating: 4,
    is_premium: true,
    preview_image: `img/apartment-01.jpg`,
    bedrooms: 3,
    max_adults: 6,
    goods: [`Wifi`, `Heating`, `Kitchen`],
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
     The building is green and from 18th century. An independent House, strategically located between Rembrand
     Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    host: {
      id: 1,
      avatar_url: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      is_pro: true,
    },
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/room.jpg`
    ],
    reviews: [
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  }, {
    id: 2,
    title: `Wood and stone place`,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    location: {
      latitude: 50.913361,
      longitude: 6.9509739999999995,
    },
    type: `Private room`,
    price: 80,
    rating: 4,
    is_premium: false,
    preview_image: `img/room.jpg`,
    bedrooms: 2,
    max_adults: 4,
    goods: [`Wifi`, `Kitchen`, `Cable TV`, `Washing machine`],
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
     The building is green and from 18th century. An independent House, strategically located between Rembrand
     Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    host: {
      id: 2,
      avatar_url: `img/avatar-max.jpg`,
      name: `Max`,
      is_pro: false,
    },
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/room.jpg`
    ],
    reviews: [
      {
        name: `Max`,
        photo: `img/avatar-max.jpg`,
        rating: 4,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
        date: `April 2019`,
      },
    ],
  }
];
