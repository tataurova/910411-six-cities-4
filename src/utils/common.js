export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const mockFunction = () => {
  // do nothing
};
