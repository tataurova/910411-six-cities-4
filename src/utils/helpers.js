export const getOfferInfo = (offers, offerId) => {
  const idNumber = Number(offerId);
  for (let key in offers) {
    if (offers.hasOwnProperty(key)) {
      const offer = offers[key].filter((item) => item.id === idNumber)[0];
      if (offer) {
        return {
          offer,
          nearOffers: offers[key],
        };
      }
    }
  }
  return {};
};
