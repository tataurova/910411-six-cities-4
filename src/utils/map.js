import leaflet from "leaflet";

export const addMarkersToMap = (offers, hoveredCardId, icon, activeIcon, map) => {
  const markers = [];
  offers.forEach((offer) => {
    const marker = leaflet.marker(offer.coordinates, {icon: offer.id === hoveredCardId ? activeIcon : icon});
    markers.push(marker);
  });
  const markersGroup = leaflet.layerGroup(markers).addTo(map);
  return markersGroup;
};
