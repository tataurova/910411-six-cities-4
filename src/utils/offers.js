export const getOfferInfo = (offers, offerId) => {
  const idNumber = Number(offerId);
  return {
    offer: offers.filter((item) => item.id === idNumber)[0],
    nearOffers: offers.filter((item) => item.id !== idNumber),
  };
  // for (let key in offers) {
  //   if (offers.hasOwnProperty(key)) {
  //     const offer = offers[key].filter((item) => item.id === idNumber)[0];
  //     if (offer) {
  //       return {
  //         offer,
  //         nearOffers: offers[key].filter((item) => item.id !== offer.id),
  //       };
  //     }
  //   }
  // }
  // return {};
};
