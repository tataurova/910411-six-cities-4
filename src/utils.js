import {SortType} from "./const.js";
import leaflet from "leaflet";

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const sortOffers = (sortType, offers) => {
  switch (sortType) {
    case SortType.DEFAULT:
      return offers;
    case SortType.PRICE_UP:
      return offers.sort((a, b) => a.price - b.price);
    case SortType.PRICE_DOWN:
      return offers.sort((a, b) => b.price - a.price);
    case SortType.RATING_DOWN:
      return offers.sort((a, b) => b.rating - a.rating);
  }
  return offers;
};

export const getSortTypeName = (sortType) => {
  switch (sortType) {
    case SortType.PRICE_UP:
      return `Price: low to high`;
    case SortType.PRICE_DOWN:
      return `Price: high to low`;
    case SortType.RATING_DOWN:
      return `Top rated first`;
  }
  return `Popular`;
};

export const getArrayOfOffers = (citiesOffers) => {
  let allOffers = [];
  Object.values(citiesOffers).forEach((item) => {
    allOffers = allOffers.concat(item);
  });
  return allOffers;
};

export const addMarkersToMap = (offers, hoveredCardId, icon, activeIcon, map) => {
  const markers = [];
  offers.forEach((offer) => {
    const marker = leaflet.marker(offer.coordinates, {icon: offer.id === hoveredCardId ? activeIcon : icon});
    markers.push(marker);
  });
  const markersGroup = leaflet.layerGroup(markers).addTo(map);
  return markersGroup;
};

export const getNearOffers = (offers, id, city) => {
  const numberId = Number(id);
  return offers.filter((offer) => offer.id !== numberId && offer.city === city);
};

export const findOfferById = (offers, id) => {
  return offers.filter((offer) => offer.id === id)[0];
};
