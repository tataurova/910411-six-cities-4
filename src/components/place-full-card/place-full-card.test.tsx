import * as React from "react";
import * as renderer from "react-test-renderer";
import PlaceFullCard from "./place-full-card";
import {BrowserRouter} from "react-router-dom";
import offers from "../../mocks/offers";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {DEFAULT_ERROR_STATUS, AuthorizationStatus} from "../../const";
import {mockFunction} from "../../utils/common";

const onBookmarkButtonCLick = jest.fn();
const mockStore = configureStore([]);
const initialState = {
  onBookmarkButtonCLick,
};
const store = mockStore(initialState);

describe(`<PlaceFullCard />`, () => {
  it(`Should PlaceFullCard render correctly`, () => {
    const id = `1`;
    const tree = renderer
      .create(
          <Provider store = {store}>
            <BrowserRouter>
              <PlaceFullCard
                offer = {offers[0]}
                id = {id}
                authorizationStatus = {AuthorizationStatus.NO_AUTH}
                user = {``}
                isFetching = {false}
                error = {DEFAULT_ERROR_STATUS}
                onSubmitForm = {mockFunction}
                reviews={[]}
                onBookmarkButtonCLick = {mockFunction}
                loadReviews = {mockFunction}
                loadNearbyOffers = {mockFunction}
                nearbyOffers = {[]}
              />
            </BrowserRouter>
          </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
