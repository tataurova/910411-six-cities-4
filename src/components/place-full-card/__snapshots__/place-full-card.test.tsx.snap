// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaceFullCard /> Should PlaceFullCard render correctly 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>,
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-03.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/studio-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Canal View Prinsengracht
            </h1>
            <button
              className="property__bookmark-button button "
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon place-card__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              6
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              120
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Wifi
              </li>
              <li
                className="property__inside-item"
              >
                Heating
              </li>
              <li
                className="property__inside-item"
              >
                Kitchen
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
     The building is green and from 18th century. An independent House, strategically located between Rembrand
     Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                0
              </span>
            </h2>
            <ul
              className="reviews__list"
            />
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      >
        <div
          id="map"
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <div
            className="cities__places-list places__list tabs__content"
          />
        </div>
      </section>
    </div>
  </main>,
]
`;
